enclave-runtime/src/lib.rs

```diff
@@ -411,16 +411,16 @@ pub unsafe extern "C" fn get_state(
 		}
 	}
 
-	if !state::exists(&shard) {
-		info!("Initialized new shard that was queried chain: {:?}", shard);
-		if let Err(e) = state::init_shard(&shard) {
-			return e.into()
-		}
-	}
-
-	let mut state = match state::load(&shard) {
+	let mut state = match state::load_initialized_state(&shard) {
 		Ok(s) => s,
-		Err(e) => return e.into(),
+		Err(e) => {
+			error!("Failed to load state: {:?}", e);
+			return if let Error::Sgx(sgx_err) = e {
```

Comment:
Does `.into()` not work here? I think it should (https://github.com/integritee-network/worker/blob/master/enclave-runtime/src/error.rs#L25)

Reply:
yes you're right, I should use that 👍